src/igmpproxy.c

```diff
@@ -140,18 +143,46 @@ int main( int ArgCn, char *ArgVc[] ) {
             break;
         }
 
+        // Open /dev/null before chrooting.
+        if (!NotAsDaemon) {
+            devnull = open("/dev/null", O_RDWR, 0);
```

Comment:
Third argument to `open` should be supplied only when `O_CREAT` or `O_TMPFILE` was specified.

Reply:
Fixed, and all tests pass.